test/appendGitlabProv.test.js

```diff
@@ -0,0 +1,103 @@
+import { strictEqual } from 'assert'
+import namespace from '@rdfjs/namespace'
+import assertThrows from 'assert-throws-async'
+import getStream from 'get-stream'
+import { isDuplex } from 'isstream'
+import { describe, it } from 'mocha'
+import rdf from 'rdf-ext'
+import { Readable } from 'readable-stream'
+import appendGitlabProv from '../lib/appendGitlabProv.js'
+import { dcat } from '../lib/namespaces.js'
+import {
+  clearMockEnvironment, setMockEnvironment
+} from './support/gitlabEnvironment.js'
+
+const ex = namespace('http://example.org/')
+const typePredicate = rdf.namedNode(
+  'http://www.w3.org/1999/02/22-rdf-syntax-ns#type')
+
+describe('metadata.appendGitlabProv', () => {
+  it('should be a factory', () => {
+    strictEqual(typeof appendGitlabProv, 'function')
+  })
+
+  it('should throw an error if no argument is given', async () => {
+    await assertThrows(async () => {
+      await appendGitlabProv()
+    }, Error, /Needs subjectsWithClass as parameter/)
+  })
+
+  it(
+    'should return a duplex stream with a subjectsWithClass (namedNode) metadata parameter',
+    async () => {
+      const step = await appendGitlabProv({
+        subjectsWithClass: dcat.Dataset
+      })
+      strictEqual(isDuplex(step), true)
+    })
+
+  it(
+    'should return a duplex stream with a subjectsWithClass (string) metadata parameter',
+    async () => {
+      const step = await appendGitlabProv({
+        subjectsWithClass: dcat.Dataset
+      })
+      strictEqual(isDuplex(step), true)
+    })
```

Comment:
These tests are identical

Besides, I would actually expect (pun intended) assertions to check that `type` equals `dcat:Dataset` in both cases